### PR TITLE
Sync with gs branch

### PR DESCRIPTION
**Pull request by ced0ps**
Aqua provided a GS specific branch upstream. I merged the changes. There apparently where other changes to the mint 5.3 branch and the 5.3_Giant_Swarm branch which where difficult to track back.

### CODE REVIEW

**Comment by webwurst** [comment]
Can you check on these linting errors please? :)
```
helm/aqua-app-server/values.yaml
  24:10     error    too few spaces before comment  (comments)
  25:30     error    too few spaces before comment  (comments)
  198:12    error    too few spaces before comment  (comments)
  198:13    error    missing starting space in comment  (comments)
  199:6     error    missing starting space in comment  (comments)
  260:17    error    trailing spaces  (trailing-spaces)
  393:97    error    trailing spaces  (trailing-spaces)
```
And is there a way to test this on one of our clusters? :thinking:

File: helm/aqua-app-enforcer/values.yaml [hunk 1]
```diff
@@ -34,18 +34,18 @@ securityContext:
...
   roleRef:
 
 gate:
-  host: aqua-gateway-svc  # for envoy deployment use host for aqua-lb and port 443
+  host: aqua-gateway-svc # for envoy deployment use host for aqua-lb and port 443
   port: 8443
 
 image:
-  repository: giantswarm/aqua-enforcer
+  repository: enforcer
```
Comment (glitchcrab) [inline helm/aqua-app-enforcer/values.yaml]: We need to retain this
```suggestion
  repository: giantswarm/aqua-enforcer
```

File: helm/aqua-app-enforcer/templates/image-pull-secret.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ metadata:
   name: {{ .Release.Name }}-registry-secret
   labels:
     app: {{ .Release.Name }}
-    chart: "{{ include "scanner.chartVersion" . }}"
+    chart: "{{ include "chartVersion" . }}"
```
Comment (glitchcrab) [inline helm/aqua-app-enforcer/templates/image-pull-secret.yaml:9]: This is generated from the helpers file: https://github.com/giantswarm/aqua-app/blob/sync_with_GS_branch/helm/aqua-app-scanner/templates/_helpers.tpl#L38

```suggestion
    chart: "{{ include "scanner.chartVersion" . }}"
```

File: helm/aqua-app-scanner/templates/scanner-deployment.yaml [hunk 3]
```diff
@@ -23,44 +23,20 @@ spec:
...
-          valueFrom:
-            secretKeyRef:
-              name: {{ .Values.secretName }}
-              key: {{ .Values.userKey }}
-        - name: SCANNER_PASSWORD
-          valueFrom:
-            secretKeyRef:
-              name: {{ .Values.secretName }}
-              key: {{ .Values.passwordKey }}
-        {{- end}}
```
Comment (glitchcrab) [inline helm/aqua-app-scanner/templates/scanner-deployment.yaml:49]: I presume this is replaced by this? https://github.com/giantswarm/aqua-app/pull/43/files?file-filters%5B%5D=.tpl&file-filters%5B%5D=.yaml#diff-c35ce371536ef76f30532c44b7cad5d8d65a10041db0e3b22eb0c37a1a730a5eR51-R53

**Comment by glitchcrab** [comment]
> And is there a way to test this on one of our clusters? thinking

Yes, I documented the minimum steps needed to test things run successfully: https://intranet.giantswarm.io/docs/organizational-structure/teams/halo/app-testing/aqua/

File: helm/aqua-app-enforcer/values.yaml (hunk 1 above)
Comment (ced0ps) [reply helm/aqua-app-enforcer/values.yaml]: damn it... I did those :facepalm:

File: helm/aqua-app-enforcer/templates/image-pull-secret.yaml (hunk 2 above)
Comment (ced0ps) [reply helm/aqua-app-enforcer/templates/image-pull-secret.yaml:9]: Yea but that's the helper for the scanner. The one for enforcer is here: https://github.com/giantswarm/aqua-app/blob/9e09f63234049ef11182792afb5be5822b5e4aa1/helm/aqua-app-enforcer/templates/_helpers.tpl#L26 right?

File: helm/aqua-app-scanner/templates/scanner-deployment.yaml (hunk 3 above)
Comment (ced0ps) [reply helm/aqua-app-scanner/templates/scanner-deployment.yaml:49]: That's my presumption as well.

File: helm/aqua-app-enforcer/templates/image-pull-secret.yaml (hunk 2 above)
Comment (glitchcrab) [reply helm/aqua-app-enforcer/templates/image-pull-secret.yaml:9]: oh yes my bad :facepalm:

File: helm/aqua-app-server/values.schema.json [hunk 4]
```diff
@@ -29,73 +29,73 @@
                     "type": "boolean"
                 },
                 "dbAuditPasswordKey": {
-                    "type": ["null", "string"]
+                    "type": "null"
```
Comment (glitchcrab) [inline helm/aqua-app-server/values.schema.json]:
```suggestion
                    "type": ["null", "string"]
```

**Comment by glitchcrab** [comment]
The values schema isn't going to work anywhere but in CI. 

All of the types which have been replaced with `"type": "null"` will fail when the chart is deployed with them set to a value. I wrote some stuff around that here https://intranet.giantswarm.io/docs/organizational-structure/teams/halo/app-updates/helm-values-schema/#null-values

You'll need to go through the values schema and make sure there isn't a single value which is only allowed to be `null`.

File: helm/aqua-app-server/values.schema.json (hunk 4 above)
Comment (glitchcrab) [reply helm/aqua-app-server/values.schema.json]: this is just an example - there shouldn't be a bare `null` type anywhere in that file

**Review by glitchcrab — APPROVED** [review]
looks good to me (assuming it deploys properly)

**Pull request merged**